StrongInject.Generator/InstanceSource.cs

```diff
@@ -118,6 +118,18 @@ public static InstanceSource Create(INamedTypeSymbol asType, InstanceSource unde
                 ? underlying
                 : new ForwardedInstanceSource(asType, underlying is ForwardedInstanceSource forwardedUnderlying ? forwardedUnderlying.Underlying : underlying);
 
+        public override void Visit<TState>(IVisitor<TState> visitor, TState state)
+        {
+            visitor.Visit(this, state);
+        }
+    }
+    internal record OwnedSource(
+        ITypeSymbol OwnedType,
+        ITypeSymbol OwnedValueType,
+        bool IsAsync) : InstanceSource(Scope.InstancePerResolution, IsAsync: IsAsync, CanDecorate: true)
```

Comment:
I think the Owned should be InstancePerDependency, and then separately we need to handle making sure the LocalFunction is deduplicated.

Reply:
Updated. I want to write some more tests before we merge, making sure IAsyncContainer is required for directly-injected Owned and AsyncOwned of things that require async initialization, making sure local functions aren't deduplicated inside to outside Func<T>, and a last sweep to think of more things.

Reply:
Should also test how StrongInject should react to people explicitly trying to register Async/Owned.

Reply:
> Should also test how StrongInject should react to people explicitly trying to register Async/Owned.

My philosophy so far is that works, and it overrides the default provision of `Owned` - so basically `Owned` just becomes a normal registration with no special behaviour.

Reply:
Okay, that makes sense. I'm not sure that falls out of the current implementation.

Reply:
Updated.